data_steward/cdr_cleaner/cleaning_rules/remove_operational_pii_fields.py

```diff
@@ -0,0 +1,239 @@
+The supplemental operational_pii_fields.csv shows all present PPI codes without a mapped concepts,
+indicating which should be dropped in the “drop_value” column
+
+Jira issues = DC-500, DC-831
+"""
+# Python imports
+import logging
+import os
+from datetime import datetime
+
+# Project imports
+from google.cloud import bigquery
+from gcloud.bq import BigQueryClient
+import resources
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule, query_spec_list
+from common import JINJA_ENV, OBSERVATION
+from typing import Dict
+
+LOGGER = logging.getLogger(__name__)
+
+OPERATIONAL_PII_FIELDS_TABLE = '_operational_pii_fields'
+INTERMEDIARY_TABLE = 'remove_operational_pii_fields_observation'
+
+JIRA_ISSUE_NUMBERS = ['DC500', 'DC831']
+
+SANDBOX_OPERATIONAL_PII_FIELDS = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE
+    `{{project_id}}.{{sandbox_dataset_id}}.{{intermediary_table}}` AS (
+  SELECT
+    *
+    FROM
+    `{{project_id}}.{{dataset_id}}.observation`
+  WHERE
+    observation_id IN (
+    SELECT
+      observation_id
+    FROM
+      `{{project_id}}.{{sandbox_dataset_id}}.{{operational_pii_fields_table}}` as pii
+    JOIN
+      `{{project_id}}.{{dataset_id}}.observation` as ob
+    USING
+        (observation_source_value)
+    WHERE
+      drop_value=TRUE)
+      )
+""")
+
+DELETE_QUERY = JINJA_ENV.from_string("""
+DELETE
+FROM
+    `{{project_id}}.{{dataset_id}}.observation`
+WHERE
+observation_id
+IN (SELECT
+    observation_id
+    FROM `{{project_id}}.{{sandbox_dataset_id}}.{{intermediary_table}}` )
+""")
+
+### Validation Query ####
+COUNTS_QUERY = JINJA_ENV.from_string("""
+SELECT
+    COUNT(*) AS total_count
+FROM
+    `{{project_id}}.{{dataset_id}}.observation`
+WHERE
+observation_id
+IN (SELECT
+    observation_id
+    FROM `{{project}}.{{sandbox_dataset}}.{{intermediary_table}}` )
+""")
+
+
+class RemoveOperationalPiiFields(BaseCleaningRule):
+
+    def __init__(self,
+                 project_id,
+                 dataset_id,
+                 sandbox_dataset_id,
+                 table_namer=None):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'Removes rows with the observation_source_values listed in the lookup table.'
+
+        super().__init__(issue_numbers=JIRA_ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[cdr_consts.RDR],
+                         affected_tables=[OBSERVATION],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         depends_on=[],
```

Comment:
This is an optional argument with a default value of depends_on=None. So, you could remove it if there is no use for it.